snuba/datasets/join_schema.py

```diff
@@ -0,0 +1,92 @@
+from dataclasses import dataclass
```

Comment:
Maybe we can create a schema package in datasets and put one schema per file cause right now it feels odd to have the schemas in just two files.

Don't need to do that in this PR but just a thought.

Reply:
I agree that `snuba.datasets` is starting to feel a little bit like a kitchen sink and getting ripe for reorganization. I think in general we haven't done a great job at intentionally creating a readable repository structure.

Reply:
I think I can move schemas in their own package here (and maybe if the planets align properly, this time there are non 0 chances that I will remember to add __init__.py).
Specifically now join_schema is on a separate file not only for better structuring but to break circular dependencies between schema and join_schema.